packages/core/src/reflection/reflection_capabilities.ts

```diff
@@ -34,15 +34,22 @@ import {GetterFn, MethodFn, SetterFn} from './types';
  *     var _this = _super.apply(this, arguments) || this;
  * ```
  *
+ * downleveled to ES5 with `downlevelIteration` for TypeScript < 4.2:
  * ```
  *   function MyClass() {
  *     var _this = _super.apply(this, __spread(arguments)) || this;
  * ```
  *
+ * or downleveled to ES5 with `downlevelIteration` for TypeScript >= 4.2:
+ * ```
+ *   function MyClass() {
+ *     var _this = _super.apply(this, __spreadArray([], __read(arguments))) || this;
+ * ```
+ *
  * More details can be found in: https://github.com/angular/angular/issues/38453.
  */
 export const ES5_DELEGATE_CTOR =
-    /^function\s+\S+\(\)\s*{[\s\S]+\.apply\(this,\s*(arguments|[^()]+\(arguments\))\)/;
+    /^function\s+\S+\(\)\s*{[\s\S]+\.apply\(this,\s*(arguments|(?:[^()]+\(\[\])?[^()]+\(arguments\))\)/;
```

Comment:
I'll leave @gkalpak to give a full regex review, but I would be tempted to pop a `,` in after the `[]`, just for good measure.

Reply:
:+1:

Adding `,` after `\[\]` seems more in spirit with the regex:

```suggestion
    /^function\s+\S+\(\)\s*{[\s\S]+\.apply\(this,\s*(arguments|(?:[^()]+\(\[\],)?[^()]+\(arguments\))\)/;
```

This regex is clearly not trying to be 100% accurate, so I am not going to try and address that :grin:

One small note:
The new regex does not check for the extra closing parenthesis at the end, so it could match, for example,

```js
function Foo() { var _this = _super.apply(this, __spreadArray([], __read(arguments)), something); }
```

But given that `.apply()` does not accept more arguments (and again given that the regex does not try to be 100% accurate), I think the regex is good enough as it is :+1:

Reply:
I like having the comma, I'll add that :+1: